src/main/java/net/neoforged/neoforge/event/entity/player/BlockInteractEvent.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.player;
+
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.common.extensions.IItemExtension;
+
+/**
+ * <p></p>Fires on both the client and server thread when a player uses an item targeting a block.
+ *
+ * <p>The event fires in three phases, corresponding with the three interaction behaviors:
+ * {@link IItemExtension#onItemUseFirst},
+ * {@link BlockBehaviour#use},
+ * and {@link Item#useOn}.</p>
+ *
+ * <p>If the event is cancelled via {@link #cancelWithResult},
+ * then the normal interaction behavior for that phase will not run,
+ * and the specified {@link InteractionResult} will be returned instead.</p>
+ */
+public class BlockInteractEvent extends PlayerInteractEvent implements ICancellableEvent
+{
+    private final UseOnContext context;
+    private final UsePhase usePhase;
+    private InteractionResult cancellationResult = InteractionResult.PASS;
+
+    public BlockInteractEvent(UseOnContext context, UsePhase usePhase)
+    {
+        super(context.getPlayer(), context.getHand(), context.getClickedPos(), context.getClickedFace());
+        this.context = context;
+        this.usePhase = usePhase;
+    }
+
+    /**
+     * @return context
+     */
+    public UseOnContext getUseOnContext()
+    {
+        return this.context;
+    }
+
+    /**
+     * @return The Use Phase of the interaction, see {@link UsePhase} for semantics
```

Comment:
```suggestion
     * {@return the Use Phase of the interaction}
```
The link should be an actual `@see` tag.